tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -510,6 +511,17 @@ private async void WidgetControl_Drop(object sender, DragEventArgs e)
         Log.Logger()?.ReportDebug("DashboardView", $"Drop ended");
     }
 
+    protected override void OnNavigatingFrom(NavigatingCancelEventArgs e)
+    {
+        Log.Logger()?.ReportDebug("DashboardView", $"Leaving Dashboard, deactivating widgets.");
+
+        // Deactivate widgets if we're not on the Dashboard.
+        foreach (var widget in PinnedWidgets)
+        {
+            widget.UnsubscribeFromWidgetUpdates();

```

Comment:
Out of curiosity, how do widgets resume after navigating to the page again. Would today's code subscribe again to the events or new widgets are created?

Reply:
From description: "we re-activate them when rebuilding the PinnedWidgets list."